vignettes/adrs.Rmd

```diff
@@ -0,0 +1,625 @@
+
+```{r}
+adsldth <- adsl %>%
+  select(STUDYID, USUBJID, DTHDT, !!!adsl_vars)
+
+adrs <- adrs %>%
+  derive_param_first_event(
+    dataset_adsl = adsldth,
+    dataset_source = adsldth,
+    filter_source = !is.na(DTHDT),
+    date_var = DTHDT,
+    set_values_to = vars(
+      PARAMCD = "DEATH",
+      PARAM = "Death",
+      PARCAT1 = "Reference Event",
+      ANL01FL = "Y"
+    )
+  ) %>%
+  select(-DTHDT)
+```
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = vars(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, ANL01FL),
+  filter = PARAMCD == "DEATH"
+)
+```
+
+## Derive Last Disease Assessment Parameters {#lsta}
+
+The function `derive_param_lasta()` can be used to create a parameter for last disease assessment. There is flexibility
+to use `source_pd` argument to only look up to first `PD` assessment, but in the below example all assessments are included.
+
+```{r}
+adrs <- adrs %>%
+  derive_param_lasta(
+    filter_source = PARAMCD == "OVR" & ANL01FL == "Y",
+    set_values_to = vars(
+      PARAMCD = "LSTA",
+      PARAM = "Last Disease Assessment by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "Recist 1.1",
+      ANL01FL = "Y"
+    )
+  )
+```
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = vars(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, ANL01FL),
+  filter = PARAMCD == "LSTA"
+)
+```
+
+## Derive Measurable Disease at Baseline Parameter {#mdis}
+
+The function `admiral::derive_param_exist_flag()` can be used to check whether a patient has measurable disease at baseline,
+according to a company-specific condition. In this example we check `TU` for target lesions during the baseline visit.
+We need to restrict the columns from `ADSL` as we'll merge all required variables later across all our `ADRS` records.
+
+```{r}
+adslmdis <- adsl %>%
+  select(STUDYID, USUBJID, !!!adsl_vars)
+
+adrs <- adrs %>%
+  derive_param_exist_flag(
+    dataset_adsl = adslmdis,
+    dataset_add = tu,
+    condition = TUEVAL == "INVESTIGATOR" & TUSTRESC == "TARGET" & VISIT == "BASELINE",
+    false_value = "N",
+    missing_value = "N",
+    set_values_to = vars(
+      PARAMCD = "MDIS",
+      PARAM = "Measurable Disease at Baseline by Investigator",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "Recist 1.1",
+      ANL01FL = "Y"
+    )
+  )
+```
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = vars(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, ANL01FL),
+  filter = PARAMCD == "MDIS"
+)
+```
+
+## Derive `ADTM`{#adtm}
+
+The function `admiral::convert_date_to_dtm()` can be used to derive `ADTM`. As our test data has
+no times collected, here we just set time as `00:00:00`.
+
+```{r eval=TRUE}
+adrs <- adrs %>%
+  mutate(
+    ADTM = convert_date_to_dtm(
+      dt = ADT,
+      date_imputation = NULL,
+      time_imputation = "00:00:00"
+    )
+  )
```

Comment:
To be ADaM compliant we need to add `ATMF = "H"`. (The Roche specs are not ADaM compliant at this point.)

However, I am not sure if we should derive `ADTM` in the vignette and the template at all. In the Roche specs it is derived for technical reasons (derivation of period related variables).

@pharmaverse/admiralonco-dev , what do you think?

Reply:
@bundfussr i'm fine with leaving it out. now removed from the vignette and template